proposals/001_resource_kubernetes_patch.md

```diff
@@ -0,0 +1,165 @@
+# Adding a `kubernetes_patch` resource 
+
+This proposal discusses adding a `kubernetes_patch` resource to the provider. 
+
+
+## Some Background
+
+[kubectl patch](https://kubernetes.io/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/) allows the user to mutate an existing API resource by specifying its name and kind, and a JSON or YAML partial with the desired change, e.g:
+
+```
+kubectl patch node k8s-node-1 -p '{"spec":{"unschedulable":true}}
+```
+
+There are 3 types of patch that can be applied:
+
+1. **strategic** – This is the default and does a Strategic Merge Patch on the resource. This means that lists in the resource will be merged with lists in the patch, combining the items. This lets you add things to lists without knowing the current contents. This type is not supported on custom resources.
```

Comment:
It might be more accurate to represent the Strategic patch as a more dynamic patch type that looks at the specific object to determine how to modify it. The behavior of the strategic patch is determined per-resource in the Kubernetes API. It will sometimes behave as 1. describes here, but sometimes it will differ.

Their docs describe it in an example [here](https://kubernetes.io/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/#notes-on-the-strategic-merge-patch) 

```
the list in the patch was merged with the existing list. This is not always what happens when you use a strategic merge patch on a list. In some cases, the list is replaced, not merged.

With a strategic merge patch, a list is either replaced or merged depending on its patch strategy
```

The API docs also contain a reference to this:

```
When patching complex types, arrays and maps, how the patch is applied is defined on a per-field basis and may either replace the field's current value, or merge the contents into the current value.
```

Here are some instances of the patchStrategy being used in the Kubernetes code:
https://github.com/kubernetes/kubernetes/search?q=patchStrategy&type=code